frontend/src/pages/storageClasses/StorageClassEnableSwitch.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { Flex, FlexItem, Switch, Tooltip } from '@patternfly/react-core';
+import { updateStorageClassConfig } from '~/services/StorageClassService';
+
+interface StorageClassEnableSwitchProps {
+  storageClassName: string;
+  isChecked: boolean;
+  isDisabled: boolean;
+  onChange: () => void;
+}
+
+export const StorageClassEnableSwitch: React.FC<StorageClassEnableSwitchProps> = ({
+  storageClassName,
+  isChecked: isInitialChecked,
+  isDisabled,
+  onChange,
+}) => {
+  const [isChecked, setIsChecked] = React.useState(isInitialChecked);
+  const [isUpdating, setIsUpdating] = React.useState(false);
+
+  const update = React.useCallback(
+    async (_event: React.FormEvent<HTMLInputElement>, checked: boolean) => {
+      setIsUpdating(true);
+
+      try {
+        await updateStorageClassConfig(storageClassName, { isEnabled: checked });
+        await onChange();
+        setIsUpdating(false);
+      } catch {
+        setIsUpdating(false);
+      }
+
+      setIsChecked(checked);
+    },
+    [onChange, storageClassName],
+  );
+
+  const switchInput = React.useMemo(
+    () => (
+      <Switch
+        id={`${storageClassName}-enable-switch`}
+        aria-label="Storage class enable switch"
+        isChecked={isChecked}
+        aria-checked={isChecked}
+        isDisabled={isDisabled || isUpdating}
+        onChange={update}
+        data-testid="enable-switch"
+      />
+    ),
+    [isChecked, isDisabled, isUpdating, storageClassName, update],
+  );
+
+  return (
+    <Flex spaceItems={{ default: 'spaceItemsMd' }} alignItems={{ default: 'alignItemsCenter' }}>
+      <FlexItem>
+        {isDisabled ? (
+          <Tooltip content="Set a new default storage class to disable this one.">
+            {switchInput}
+          </Tooltip>
+        ) : (
+          switchInput
+        )}
+      </FlexItem>
+
+      <FlexItem
+        className="pf-v5-u-disabled-color-100"
+        style={{ visibility: isUpdating ? 'visible' : 'hidden' }}
+      >
+        {isChecked ? 'Disabling' : 'Enabling'}...
```

Comment:
to follow what is done in other areas: disable the checkbox when loading, not need for text

Reply:
This was specifically asked for here by @xianli123

https://www.figma.com/proto/372he4ynUVkovm3lwfFVbg/Storage-classes?page-id=978%3A58598&node-id=978-58663&node-type=FRAME&viewport=808%2C369%2C0.09&t=ZZSwrwX1rXaSAepA-1&scaling=min-zoom&content-scaling=fixed

Reply:
Hi @Gkrumbach07 @jpuzz0 This pattern refers to what we have in the workbench. I attached a screenshot below. 
When stopping a workbench, there will be a transition phase named 'Stopping...'. We can reuse this pattern. Does it make sense?
<img width="1423" alt="Screenshot 2024-09-05 at 14 46 21" src="https://github.com/user-attachments/assets/856082b2-f78a-43b7-b9ad-827afa509530">

Reply:
Ope i stand corrected. carry on then